### PR TITLE
feat(material/core): add typography config normalization function

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/material/core/typography/_typography.scss [hunk 1]
```diff
@@ -69,6 +69,37 @@
   @return map-merge($config, (font-family: $font-family));
 }
 
+// Whether a config is for the Material Design 2018 typography system.
+@function mat-private-typography-is-2018-config($config) {
+  @if map-get($config, headline-1) {
+    @return true;
+  }
+  @return false;
```
Comment (crisbeto) [inline src/material/core/typography/_typography.scss:77]: Could we use `@return map-get($config, headline-1) != null` or is that just not how Sass works?

File: src/material/core/typography/_typography.scss [hunk 2]
```diff
@@ -69,6 +69,37 @@
...
+}
+
+// Given a config for either the 2014 or 2018 Material Design typography system,
+// produces a normalized typography config for the 2014 Material Design typography system.
+// 2014 - https://material.io/archive/guidelines/style/typography.html#typography-styles
+// 2018 - https://material.io/design/typography/the-type-system.html#type-scale
+@function mat-typography-normalized-config($config) {
```
Comment (crisbeto) [inline src/material/core/typography/_typography.scss:84]: Could we bake this into `mat-typography` config and have it convert automatically. We could only do it if we can determine with some degree of certainty it's an actual 2018 config.

Comment (mmalerba) [reply src/material/core/typography/_typography.scss:84]: We need this to be separate as we plan to use it to repair broken configs that are accidentally generated by some code inside Google.

File: src/material/core/typography/_typography.scss (hunk 1 above)
Comment (mmalerba) [reply src/material/core/typography/_typography.scss:77]: It looks like you can do that

File: src/material/core/typography/_typography.scss (hunk 2 above)
Comment (crisbeto) [reply src/material/core/typography/_typography.scss:81]: My hope was that we'd be able to slowly transition external apps to the 2018 config, but that can always be done later on.

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**